### PR TITLE
Second milestone: User-Interface

### PR DESCRIPTION
**Pull request by EdCrux**
The following path for this milestone: 

We create 6 files that contain: 
- Within the View module, we have an array that keeps in the logo, a function to create new lines and a function to create a line to divide the main section of the logo
- Welcome class print the logo and a welcome message, also clear the page each time you input something.
- Names class helps to print a message to the user and store the name. 
- Board, renders the board
- Input tracks the input of each move

### CODE REVIEW

File: lib/input.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
+require_relative '../lib/board.rb'
+require_relative '../lib/names.rb'
+require_relative '../lib/welcome.rb'
+
+class Input
+  def ask_for_move(players)
+    br_arr = %w[1 2 3 4 5 6 7 8 9]
+    9.times do |x|
```
Comment (mosesogwo) [inline lib/input.rb]: Use a loop that terminates when there is a win or a draw instead of a fixed number of times.

**Review by mosesogwo — CHANGES_REQUESTED** [review]
**HI @EdCrux , @mariancristu, Nice job...:clap:, but let's take care of the following:**

1. Don't edit linter config files. Undo the changes made to **.rubocop.yml** file.

2. Your `puts/prints` and `get.chomps` commands should be used in main.rb only.



3. You should inform the user when there is a draw or a win.

4. **Optional:** I think the suggested branch_name **user_interface** describes the branch better than **Second Milestone**. You should consider renaming the branch.



_**Kindly make these changes and submit another code review request.**_

**Any question? You can reach me on Slack (Moses Ogwo)**

**Cheers:stars:**

### Status: Changes requested

File: lib/input.rb [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      Welcome.clear
+      Welcome.print_message(print, &print)
+      Board.render(br_arr, &print)
+      (x % 2).zero? ? action(players[0], br_arr, 'X', get_value, &print) : action(players[1], br_arr, 'O', get_value, &print)
+      break if winner
+
+      Board.render(br_arr, &print) if x == 8
+    end
+  end
```
Comment (stratospherique) [inline lib/input.rb]:
```suggestion
    while game_on
      Welcome.clear
      Welcome.print_message(print, &print)
      Board.render(br_arr, &print)
      (x % 2).zero? ? action(players[0], br_arr, 'X', get_value, &print) : action(players[1], br_arr, 'O', get_value, &print)
      
      game_on = false if winner || draw
      Board.render(br_arr, &print) if x == 8
    end
  end
```

**Review by stratospherique — CHANGES_REQUESTED** [review]
### Great progress @team :clap: However, some of the changes requested by @mosesogwo were not implemented. Check :arrow_down: 

* [x] You shouldn't edit linter config files. Undo the changes made to .rubocop.yml file.

* [ ] User Interaction related code should reside in bin folder. :pray: move code that uses puts/print get.chomp, etc. to bin folder.

### That's it :relieved: When finished, kindly resubmit another code review request :pray: 

Happy coding :v: 
**_Ahmed Mahfoudh_**

**Comment by Anna-Myzukina** [comment]
## Hi team !  

# Status: [changes required]


* Good job so far, just a couple more changes and your project can be approved!


* [ ] `main.rb` uses `puts/print` and `get.chomp` commands.  From [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))screenshot
## [OPTIONAL] 
Using `require_relative` for files in your directory instead of `require` is a ruby convention.

> "Require should be used for external files, like gems, while require_relative should be used for referring to files within your directory."


### Make changes and submit another code review request.

*If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you.*
*Happy coding * <g-emoji class="g-emoji" alias="computer" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4bb.png">computer</g-emoji>

**Review by whiz25 — APPROVED** [review]
## Congz :clap: 
You have made the requested changes and therefore it's a :+1: from me. Merge and proceed.

### Martin Afani

**Pull request merged**